File: BackEnd/src/main/java/com/example/demo/models/Customer.java
package com.example.demo.models;
import com.example.demo.embedded.Person;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;
import java.util.List;

@Entity
@Data
@Table(name = "customer")
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String email;
    private String phone;
    @Embedded
    private Person generaleInfo;
    @OneToMany(mappedBy = "customer")
    private List<Construction> constructions;
}

File: FrontEnd/src/app/services/models/customer.ts
/* tslint:disable */
/* eslint-disable */
import { Construction } from '../models/construction';
import { Person } from '../models/person';
export interface Customer {
  constructions: Array<Construction>;
  email: string;
  generaleInfo: Person;
  id: number;
  phone: string;
}
